### PR TITLE
New firmware release script

### PR DESCRIPTION
**Pull request by slacrherbst**
This script will allow the user to create a firmware release with release notes and attached files. I have also updated the releaseNotes script to serve as a library as well as a executable.

No support for cpsw yet.

Example command line:

`````
python ../submodules/ruckus/release_gen.py --project=~/projects/smurf/cryo-det --push --version=v1.2.14 --user=slacrherbst --password=xxxxxx --prev=v1.2.13 
`````
Args can either be passed on the command line or the user will be prompted for input.

A zip file is generated with rogue python files, configuration files and images. The selected images are also attached to the release. Example release:

https://github.com/slaclab/cryo-det/releases/tag/MicrowaveMuxBpEthGen2_v1.2.14

Example release configuration file:

 `````
CommonPython:
   - python
   - submodules/amc-carrier-core/python
   - submodules/axi-pcie-core/python
   - submodules/lcls-timing-core/python
   - submodules/surf/python

CommonConfig:

TopPackage: CryoDet

Targets:
   MicrowaveMuxBpEthGen2:
      Extensions:
      - mcs.gz

Releases:
   MicrowaveMuxBpEthGen2:
      Targets:
      - MicrowaveMuxBpEthGen2
      Types:
      - Rogue
      - CPSW
      Config:
      - targets/MicrowaveMuxBpEthGen2/config
      Python:
`````

### CODE REVIEW

File: release_gen.py [hunk 1]
```diff
@@ -0,0 +1,381 @@
+#!/usr/bin/env python3
+
```
Comment (ruck314) [inline release_gen.py]: There should be some documentation (or pointer to confluence page) explaining what this script does and how to use it.

Comment (ruck314) [inline release_gen.py]: You are missing the SLAC licensing header

File: release_gen.py [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+import os
+import yaml
+import argparse
+import zipfile
+
+import git    # GitPython
+import github # PyGithub
```
Comment (ruck314) [inline release_gen.py]: Should pip install github be included in our rogue anaconda receipt?

**Comment by ruck314** [comment]
@slacrherbst: I did a `stress test` of the releaseNotes.py and failed on `Author`:
```
firmware/submodules/ruckus$ python3 releaseNotes.py v1.8.0...v1.0.0
Using range: v1.8.0...v1.0.0
Enter your github token: 78a2cbe630c8512dec2044664710562a374307c8
Traceback (most recent call last):
  File "releaseNotes.py", line 176, in <module>
    md = getReleaseNotes(g,repo,tags)
  File "releaseNotes.py", line 96, in getReleaseNotes
    if entry[i] is not None:
KeyError: 'Author'
```

**Comment by slacrherbst** [comment]
That error is weird. Is that script active? I assumed it was and just changed it structure a bit, but maybe there is something else wrong.

File: release_gen.py (hunk 2 above)
Comment (slacrherbst) [reply release_gen.py]: Maybe but that would not help here since the user will not have a rogue environment when building in xilinx.

**Comment by slacrherbst** [comment]
I have added cpsw support. I am also removing the zipfile generation from ruckus and removing the auto tag generation. The rest we will leave for now.

File: release_gen.py (hunk 1 above)
Comment (ruck314) [reply release_gen.py]: Resolved in 1c27774

Comment (ruck314) [reply release_gen.py]: Added an empty "description" field in the code header to be filled out

**Comment by ruck314** [comment]
Confirmed CPSW behavior has not changed.

**Pull request merged**